swap/src/network/request_response.rs

```diff
@@ -19,54 +17,23 @@ pub const TIMEOUT: u64 = 3600; // One hour.
 /// Message receive buffer.
 pub const BUF_SIZE: usize = 1024 * 1024;
 
-// TODO: Think about whether there is a better way to do this, e.g., separate
-// Codec for each Message and a macro that implements them.
-
-/// Messages Bob sends to Alice.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum BobToAlice {
-    SwapRequest(Box<bob::SwapRequest>),
-    Message0(Box<Message0>),
-    Message2(Box<Message2>),
-    Message4(Box<Message4>),
-}
-
-/// Messages Alice sends to Bob.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum AliceToBob {
-    SwapResponse(Box<alice::SwapResponse>),
-    Message1(Box<Message1>),
-    Message3(Box<Message3>),
-    Message2,
-}
-
-/// Messages sent from one party to the other.
-/// All responses are empty
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum Request {
+    SwapRequest(Box<SwapRequest>),
     TransferProof(Box<TransferProof>),
     EncryptedSignature(Box<EncryptedSignature>),
 }
 
-#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
-/// Response are only used for acknowledgement purposes.
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum Response {
+    SwapResponse(Box<SwapResponse>),
     TransferProof,
     EncryptedSignature,
 }
```

Comment:
This model looks like it can be improved. I doubt that a `TransferProof` request can be answered with an `EncryptedSignature` response. Why are these bundled together into one enum?

Reply:
> Why are these bundled together into one enum?

Inspired for existing design, I haven't yet dug enough in the code to understand if there was any reason for this design and whether it can be split.